josh-works/terakeet_books
# Schema finished - tests pass

# High level overview

I implemented the following schema, with:

- models
- migrations
- rspec tests for associations and validations
- FactoryBot and a seeds file to build real data

![picture of built schema](https://cl.ly/442r0F1W1v2G/WWW_SQL_Designer_-_default.jpg)

# To test

clone down and checkout the branch:

```bash
$ git clone git@github.com:josh-works/terakeet_books.git
$ git checkout origin/build_schema
```
and then set up the app:

```bash
$ bundle
$ rake db:create
$ rake db:migrate
```

and run the tests!

```bash
$ rspec
```

and you should see:

![passing tests](https://cl.ly/1A2O0X341O1I/1__joshthompson_Joshs-MBP-236____workspace_terakeet_books__zsh_.jpg)

I'm going to merge this into master, and then build out the final functionality.
